test/helpers/host_groups_helper_test.rb

```diff
@@ -7,6 +7,12 @@ class HostGroupsHelperTest < ActionView::TestCase
   include HostsHelper
   include ::FormHelper
 
+  # authorized_for is set as helper_method when
+  # Foreman::Controller::Authorize is included which
+  # causing "SystemStackError: stack level too deep"
+  # to avoid this I am using prepend.
+  prepend Foreman::Controller::Authorize
```

Comment:
Oh, that's some hack. That won't do. Why do you need this here?

Reply:
`label_with_link` is calling `link_to_if_authorized` which is calling `authorized_for` which is available from the `ApplicationController`.

Reply:
I'm wondering why you include this in a test and how this helps to fix the issue.

Reply:
without it I get `NoMethod: "authorized_for" ` exception

Reply:
What do you think about this?
https://github.com/timogoebel/foreman/commit/90758ac3a2743fed504d462859a2105806069352

It delegates these methods to the helper scope. It has the advantage, that the helper methods are executed in the view scope and not the controller.

Reply:
LGTM